deps/thpool/thpool.c

```diff
@@ -450,7 +442,7 @@ static void* thread_do(struct thread* thread_p) {
   while (thpool_p->keepalive) {
 
     bsem_wait(thpool_p->jobqueue.has_jobs);
-
+    LOG_IF_EXISTS("debug", "Thread-%d is running iteration", thread_p->id)
     if (thpool_p->keepalive) {
```

Comment:
Why is it a thread's responsibility to end the thread pool?

jobs = 2; thpool_p->num_threads_working = 0
thread 1: thpool_p->num_threads_working ++; < do job>; thpool_p->num_threads_working--; bsem_post_all
sem = bsem_p->total_threads_count

jobs = 1; thpool_p->num_threads_working = 0
thread 1: thpool_p->num_threads_working ++; < do job>; thpool_p->num_threads_working--; bsem_post_all
sem = 2 * bsem_p->total_threads_count